features/vaadin-node-maps/src/main/java/org/opennms/features/vaadin/nodemaps/internal/HttpServletRequestVaadinImpl.java

```diff
@@ -449,11 +418,26 @@ public long getContentLengthLong() {
 
     @Override
     public String changeSessionId() {
-        throw new RuntimeException("Cannot change session id");
+        return getServletRequest().get().changeSessionId();
     }
 
     @Override
-    public <T extends HttpUpgradeHandler> T upgrade(Class<T> handlerClass) throws IOException, ServletException {
-        throw new RuntimeException("Cannot upgrade.");
+    public <T extends HttpUpgradeHandler> T upgrade(final Class<T> handlerClass) throws IOException, ServletException {
+        return getServletRequest().get().upgrade(handlerClass);
+    }
+
+    private Optional<VaadinServletRequest> getServletRequest() {
+        if (m_request instanceof VaadinServletRequest) {
+            return Optional.ofNullable((VaadinServletRequest)m_request);
+        }
+        return Optional.empty();
```

Comment:
I am not sure if this is reasonable. Looking at the code as is, `getServletRequest().get()....` may throw a `NullPointerException`. I think in this case we should just throw an Exception if the request is not a `VaadinServletRequest`?

Reply:
I just figured if something really weird happens, we want to just let it throw an exception anyways, and there isn't much point in trying to roll it up into some other exception.

Reply:
I also think we would be better off throwing an exception if the we cannot cast the request to a `VaadinServletRequest` - we currently return an optional, but none of the callers treat it as optional.

Maybe just throw a IllegalStateException?

Reply:
Actually we don't need the wrapper at all. See last commit from @RangerRick (https://github.com/OpenNMS/opennms/pull/2144/commits/b5ee9ca277db35be12a55e55a83c18350b247a07). We can just use the same code as the `TopologyUI` is doing :)